### ISSUE TITLE
Generic static extension method can't be recognized by IDEA

### ISSUE THREAD

**Issue by mizhoux** [manifold-systems/manifold#409]
**Describe the bug**
Generic static extension method can't be recognized by IDEA.

**To Reproduce**

`MyListExtensions` is the extension class of `java.util.List`:

```java
@Extension
public class MyListExtensions {

    @Extension
    public static <E> List<E> empty() {
        return Collections.emptyList();
    }

    @Extension
    public static <E> List<E> of(E e) {
        return Collections.singletonList(e);
    }
}
```

In the App.java, method `getKeyWords` calls these extension methods:

```java
public class App {

    public static void main(String[] args) {
        List<String> keyWords = getKeyWords(3);
        System.out.println(keyWords);
    }

    public static List<String> getKeyWords(int input) {
        if (input == 0) {
            // fine
            return Collections.emptyList();
        }

        if (input == 1) {
            // fine
            return MyListExtensions.empty();
        }

        if (input == 2) {
            // got an error
            return List.empty();
        }

        // fine
        return List.of("someKeyword");
    }
}
```

Then I got an error from IDEA:

<img width="985" alt="error" src="https://user-images.githubusercontent.com/15724596/208629349-688b5141-6689-4089-baf7-ba351264bba9.png">

But the project can compile and run successfully. 

**Expected behavior**
Generic static extension method can be recognized by IDEA correctly.

**Desktop (please complete the following information):**
- OS Type & Version: macOS Big Sur 11.7.1
- Java/JDK version: 1.0.8.321
- IDE version (IntelliJ IDEA or Android Studio): 2022.3
- Manifold version: 2022.1.29
- Manifold IntelliJ plugin version: 2022.3.29

**Comment by rsmckinney** [comment]
Fixed with manifold intellij plugin releases 2022.2.31, 2022.3.32

**Comment by mizhoux** [comment]
Nice job!
